lib/Dialect/FIRRTL/FIRRTLOps.cpp

```diff
@@ -253,16 +253,16 @@ static void buildModule(OpBuilder &builder, OperationState &result,
 
   // Record the names of the arguments if present.
   SmallString<8> attrNameBuf;
-  for (size_t i = 0, e = ports.size(); i != e; ++i) {
+  // Record the names of the arguments if present.
+  SmallVector<Attribute> argNames;
+  for (size_t i = 0, e = ports.size(); i != e; ++i)
     if (ports[i].getName().empty())
-      continue;
-
-    auto argAttr =
-        NamedAttribute(builder.getIdentifier("firrtl.name"), ports[i].name);
+      argNames.push_back(builder.getStringAttr(""));
+    else
+      argNames.push_back(ports[i].name);
 
-    result.addAttribute(getArgAttrName(i, attrNameBuf),
-                        builder.getDictionaryAttr(argAttr));
-  }
+  if (!argNames.empty())
+    result.addAttribute("argNames", builder.getArrayAttr(argNames));
```

Comment:
I guess this wasn't an entirely accurate statement, but I think we should unconditionally add the argNames attribute.  Is there a reason it's structured this way? (e.g. to avoid putting an empty attribute on an external modules?)

Reply:
port names makes sense for external modules too.  Should the verifier check that argNames exists and is long enough?